tt_metal/include/tt_metal/buffer.hpp

```diff
@@ -3,46 +3,38 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #pragma once
-#include "types.hpp"
+
 #include "tt_metal/impl/buffers/buffer.hpp"
+#include "tt_metal/types.hpp"
 //==================================================
 //                  BUFFER HANDLING
 //==================================================
 
-namespace tt::tt_metal{
+namespace tt::tt_metal {
 namespace v1 {
 
-
 /**
  * @brief Allocates an interleaved DRAM or L1 buffer on the device.
  *
  * @param config Configuration for the buffer.
  * @return Buffer handle to the allocated buffer.
  */
-Buffer CreateBuffer(const InterleavedBufferConfig &config);
-
-/**
- * @brief Allocates a buffer on the device.
- *
- * @param buffer The buffer to allocate.
- * @param bottom_up If true, allocates buffer from the bottom up.
- */
-void AllocateBuffer(Buffer buffer, bool bottom_up);
+BufferHandle CreateBuffer(InterleavedBufferConfig config);
```

Comment:
What happened to `AllocateBuffer`? Is it collapsed under `CreateBuffer` now?

Reply:
Good catch, I'd forgotten I removed this. I had some difficulty in figuring out where the `bottom_up` parameter came from, and as far as I could tell, there was nothing in the v0 API that would allow us to `AllocateBuffer` with `bottom_up` specified. If we remove that parameter, I can re-add it and delegate to the v0 `detail::AllocateBuffer`, but it appears that `CreateBuffer` which eventually delegates to `Buffer::create` (which does take a `bottom_up` parameter) pushes a `detail::AllocateBuffer` call onto the worker queue anyway, so I wasn't sure whether this needed to continue to exist in the API.

Reply:
Yeah, you got it. From an API point of view, our `CreateBuffer` currently already implicitly makes a call to `AllocateBuffer`. If this is the behavior we want to retain, is there any point in decoupling `CreateBuffer` from `AllocateBuffer`? 

I'm assuming the compiler folks don't use the `CreateBuffer` API and instead default to accessing the `Buffer::create` directly if they want manual control over addresses and allocations. In that case, is that a gap at the API level?

Reply:
I don't have enough information to answer this question with complete confidence right now, but we can always add back `AllocateBuffer` later if we find that it is a gap, but once we add it, it's harder to remove if we later determine that it doesn't make sense to keep at the API level. So I'm inclined to suggest we leave it out for now.